### PR TITLE
Switch to src layout

### PR DESCRIPTION
**Pull request by samet-akcay**
# Description

- This PR changes the project structure to src layout. The overall tree structure will be as follows:

```bash
├── CHANGELOG.md
├── CITATION.cff
├── CODE_OF_CONDUCT.md
├── CONTRIBUTING.md
├── datasets
├── docs
│   ├── anomalib.png
│   ├── Makefile
│   └── source
├── LICENSE
├── MANIFEST.in
├── notebooks
│   ├── 000_getting_started
│   ├── 100_datamodules
│   ├── 200_models
│   ├── 300_benchmarking
│   ├── 400_openvino
│   └── README.md
├── pyproject.toml
├── README.md
├── requirements
│   ├── base.txt
│   ├── dev.txt
│   ├── docs.txt
│   ├── extras.txt
│   └── openvino.txt
├── setup.py
├── src
│   └── anomalib
├── tests
│   ├── helpers
│   ├── __init__.py
│   ├── nightly
│   └── pre_merge
├── third-party-programs.txt
├── tools
│   ├── benchmarking
│   ├── hpo
│   ├── inference
│   ├── test.py
│   └── train.py
└── tox.ini
```

This official python [link](https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/#:~:text=The%20src%20layout%20requires%20installation,installation%20is%20used%20for%20testing) outlines the main advantages of this layout. Some of these are as follows:

> - The src layout requires installation of the project to be able to run its code, and the flat layout does not.
>This means that the src layout involves an additional step in the development workflow of a project (typically, an [editable installation](https://setuptools.pypa.io/en/latest/userguide/development_mode.html) is used for development and a regular installation is used for testing).
>
> - The src layout helps prevent accidental usage of the in-development copy of the code.
This is relevant since the Python interpreter includes the current working directory as the first item on the import path. This means that if an import package exists in the current working directory with the same name as an installed import package, the variant from the current working directory will be used. This can lead to subtle misconfiguration of the project’s packaging tooling, which could result in files not being included in a distribution.
>
> - The src layout helps avoid this by keeping import packages in a directory separate from the root directory of the project, ensuring that the installed copy is used.
>
> - The src layout helps enforce that an [editable installation](https://setuptools.pypa.io/en/latest/userguide/development_mode.html) is only able to import files that were meant to be importable.
This is especially relevant when the editable installation is implemented using a [path configuration file](https://docs.python.org/3/library/site.html#index-2) that adds the directory to the import path.
The flat layout would add the other project files (eg: README.md, tox.ini) and packaging/tooling configuration files (eg: setup.py, noxfile.py) on the import path. This would make certain imports work in editable installations but not regular installations.

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the [pre-commit style and check guidelines](https://github.com/openvinotoolkit/anomalib/blob/main/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -73,6 +73,18 @@ follow_imports = "skip"
 follow_imports_for_stubs = true
 
 
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+# PYTEST CONFIGURATION                                                        #
```
Comment (djdameln) [inline pyproject.toml:77]: Are the changes to this file related to switching to src layout or is this something separate?

File: tox.ini [hunk 2]
```diff
@@ -32,7 +32,7 @@ deps =
 commands =
     ; 1. Run Coverage First.
     coverage erase
-    coverage run --include=anomalib/* -m pytest {posargs:tests/pre_merge/} -ra --showlocals
+    coverage run --include=anomalib/* -m pytest {posargs:tests/pre_merge/}
```
Comment (ashwinvaidya17) [inline tox.ini:35]: How does this work now? Should it be `--include=src/anomalib/*`

**Review by ashwinvaidya17 — COMMENTED** [review]
Thanks. One question

File: pyproject.toml (hunk 1 above)
Comment (samet-akcay) [reply pyproject.toml:77]: it's separate, but would be good to have. VSCode would automatically detect the tests

File: tox.ini (hunk 2 above)
Comment (samet-akcay) [reply tox.ini:35]: Let me double check

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks

**Pull request merged**